finetuner/tuner/pytorch/datasets.py

```diff
@@ -1,12 +1,11 @@
-from torch.utils.data import IterableDataset
+from torch.utils.data import Dataset
 
-from ..base import BaseDataset
-from ..dataset import SiameseMixin, TripletMixin
+from ..dataset import ClassDataset, SessionDataset
 
 
-class SiameseDataset(SiameseMixin, BaseDataset, IterableDataset):
+class PytorchClassDataset(ClassDataset, Dataset):
     ...
 
 
-class TripletDataset(TripletMixin, BaseDataset, IterableDataset):
+class PytorchSessionDataset(SessionDataset, Dataset):
```

Comment:
why is this neederd? Why is `DataSet` and `SessionDataSet` orthogonal and needed to be mixed in?

Reply:
It seems like a big smell

Reply:
This is due to pytorch DataLoader. It requires the dataset input to it to subclass its `Dataset`, see [here](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader).

It actually works without that (because `Dataset` in essence only implements `__getitem__`), but that is not documented, so I want to properly subclass `Dataset` to avoid any "suprises" in the future

Reply:
```suggestion
class PytorchSessionDataset(SessionDataset, PytorchDataset):
```